spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateV3SpecTest.java

```diff
@@ -0,0 +1,143 @@
+    public void shouldEncodeManyContentUrls() throws Exception {
+        ProductDTO product = Products.randomSKU();
+        product.setName(StringUtil.random("Test Product"));
+        product.addAttributesItem(new AttributeDTO().name("version").value("6.4"));
+        product.addAttributesItem(new AttributeDTO().name("arch").value("i386, x86_64"));
+        product.addAttributesItem(new AttributeDTO().name("sockets").value("4"));
+        product.addAttributesItem(new AttributeDTO().name("cores").value("8"));
+        product.addAttributesItem(new AttributeDTO().name("ram").value("16"));
+        product.addAttributesItem(new AttributeDTO().name("usage").value("Disaster Recovery"));
+        product.addAttributesItem(new AttributeDTO().name("roles")
+            .value("Red Hat Enterprise Linux Server, Red Hat Enterprise Linux Workstation"));
+        product.addAttributesItem(new AttributeDTO().name("addons")
+            .value("my_server_addon, my_workstation_addon"));
+        product.addAttributesItem(new AttributeDTO().name("warning_period").value("15"));
+        product.addAttributesItem(new AttributeDTO().name("management_enabled").value("true"));
+        product.addAttributesItem(new AttributeDTO().name("stacking_id").value("8888"));
+        product.addAttributesItem(new AttributeDTO().name("virt_only").value("false"));
+        product.addAttributesItem(new AttributeDTO().name("support_level").value("standard"));
+        product.addAttributesItem(new AttributeDTO().name("support_type").value("excellent"));
+        product = ownerProductApi.createProductByOwner(owner.getKey(), product);
+
+        for (int i = 0; i < 100; i++) {
+            ContentDTO content = Contents.random();
+            content.setContentUrl(
+                String.format("/content/dist/rhel/$releasever-%s/$basearch-%s/debug-%s", i, i, i));
+            content = ownerContentApi.createContent(owner.getKey(), content);
+            product = ownerProductApi.addContent(owner.getKey(), product.getId(), content.getId(), true);
+        }
+
+        PoolDTO pool = new PoolDTO();
+        pool.setProductId(product.getId());
+        pool.setStartDate(Instant.now().atOffset(ZoneOffset.UTC));
+        pool.setEndDate(Instant.now().plus(365, ChronoUnit.DAYS).atOffset(ZoneOffset.UTC));
+        pool = ownerApi.createPool(owner.getKey(), pool);
+
+        AsyncJobStatusDTO refresh = ownerApi.refreshPools(owner.getKey(), false);
+        if (refresh != null) {
+            client.jobs().waitForJobToComplete(refresh.getId(), 15000);
+        }
+        ConsumerDTO consumer = client.consumers().register(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.trustedConsumer(consumer.getUuid());
+        consumerApi = consumerClient.consumers();
+        String bindResult = consumerApi.bind(consumer.getUuid(),
+            pool.getId(), null, null, null, null, false, null, null);
+        JsonNode result = new ObjectMapper().readTree(bindResult);
```

Comment:
If we are not going to return DTO, we should at least return JsonNode so that spec tests don't have to handle their own object mappers.